### ISSUE TITLE
Do not show the eye icon for visible nodes in the Scene Tree view

### ISSUE THREAD

**Issue by npinsker** [godotengine/godot-proposals#10911]
### Describe the project you are working on

Games

### Describe the problem or limitation you are having in your project

<img src="https://github.com/user-attachments/assets/10e86081-36d5-48b4-91e6-812994c0c387" width="150" />

There is an eye symbol next to (almost) every Node in the Scene Tree view. When there are lots of nodes, this means lots of symbols. They draw the eye, and contribute (a little) to making the editor feel more cluttered.

Toggling visibility isn't so common that it deserves such prominent placement in the engine, and we could remove the eye icons and clean up the look a little bit. When needed, toggling visibility can be done via the Inspector, which will be made even easier with the upcoming https://github.com/godotengine/godot/pull/97415.

If that's too cumbersome, alternate solutions could involve adding a toggle to the right-click menu for that node, or follow Unity's lead and providing a top-level checkbox in the Inspector:
<img src="https://github.com/user-attachments/assets/41f8232f-5a5c-4939-80dd-180b8ef6f6c3" width="300" />

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Remove the eye icon for visible nodes. (Invisible nodes will be unaffected)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

- If a node is visible, show no eye icon.
- If a node is invisible, show the closed eye icon, as before.
- (possibly) Add an option to the right-click context menu to toggle visibility.

Visibility can already be changed through the Inspector.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Will be used constantly, in that it will simplify the look of the editor

### Is there a reason why this should be core and not an add-on in the asset library?

It affects every project

**Comment by npinsker** [comment]
FWIW: Unity & Unreal Engine 5 use color (white/gray) to denote visibility. Unreal 5 has an eye icon that's only displayed on hover. Unreal Engine 4 & Game Maker use always-visible eye icons.

**Comment by Calinou** [comment]
We currently don't have a system in place to only show specific TreeItem icons on hover, so this would have to be implemented first. See also https://github.com/godotengine/godot-proposals/issues/7160.

**Comment by npinsker** [comment]
Hovering would also be a more complicated UX problem here because Unreal has a dedicated column for visibility, while the most naive Godot impl would push the other right-aligned icons around in an awkward way.

**Comment by Calinou** [comment]
If we don't show a visible eye icon on hover, then discoverability will suffer unless you try to click in the invisible area. I assume it would allow toggling visibility even if invisible, since the closed eye icon would appear there if invisible. It would also still take up horizontal space in the Scene tree dock.

**Comment by KoBeWi** [comment]
> We currently don't have a system in place to only show specific TreeItem icons on hover, so this would have to be implemented first.

Not really, it's possible to add/remove buttons on hover manually. Somewhat hacky, but it should be doable already.

**Comment by Calinou** [comment]
> Not really, it's possible to add/remove buttons on hover manually. Somewhat hacky, but it should be doable already.

Won't this impact performance in large trees due to constant tree rebuilds when the hovered TreeItem changes?

**Comment by KoBeWi** [comment]
Modifying buttons does not rebuild tree.

**Comment by davthedev** [comment]
I do use this visibility toggle quite often on different items to quickly see what's underneath, such as tilemap layers. Would prefer keeping the icon accessible at all times.

The reveal-on-hover approach is not something I would go for, personally. This is inherited from the web and hinders discoverability. It also makes a surprise change in the course of a quick action. For example, a user would want to select a node, target the tree row in a quick motion and -surprise!- click the visibility instead. 

Another example, for those who know a bit about working in audio, is that in DAWs (the software used to make music), there are different tracks, each one having their mute/solo buttons easily accessible because this is something you keep toggling on and off all the time in some workflows. The visibility icon is the "mute" equivalent in scene editing, especially for 2D. Its position should be consistent, which sets the case against reveal-on-hover that causes shifts.

If it annoys users, I would suggest making it a setting instead of removing it for everyone. I need this feature.

Regarding the technical cost of changing the icon display dynamically, this is not costly at all.
The drawing is done by old-fashioned painting code, which intelligently renders what is within the visibility window only (an art that is quietly being lost!). There is no such thing as a DOM-like heavy lifting. No React, no complex state recalculation here, no memory model for each single string and rectangle! The data structure focuses on the essentials. If something changes, everything is repainted.
